src/rpm/package.rs

```diff
@@ -591,9 +591,11 @@ impl RPMPackageMetadata {
     }
 
     #[inline]
-    pub fn get_payload_compressor(&self) -> Result<&str, RPMError> {
-        self.header
-            .get_entry_data_as_string(IndexTag::RPMTAG_PAYLOADCOMPRESSOR)
+    pub fn get_payload_compressor(&self) -> Result<CompressionType, RPMError> {
+        let comp_str = self
```

Comment:
If this is the only place where `CompressionType::from_str()` is used then lets inline it here.

In the case of an uncompressed package, the tag simply won't exist.  So we could either return the error, or match on the result and return `CompressionType::None` in that case.  @drahnr What do you prefer?

Reply:
@dralley I suggest something like this:

```
enum CompressionType {
    None, 
    ...,
    Unknown(String)
```

What do you think about this variant?

Reply:
@ikrivosheev I don't have a strong opinion, I am against inlining `FromStr` though.